lms/views/lti/basic_launch.py

```diff
@@ -298,19 +300,23 @@ def _do_launch(self, document_url, grading_supported=True):
     def _store_lti_data(self):
         """Store LTI launch data in our LMS database."""
 
-        request = self.request
+        self.application_instance.update_lms_data(self.context.lti_params)
+
+        lti_params = self.context.lti_params
 
         # Before any LTI assignments launch, create or update the Hypothesis
         # user and group corresponding to the LTI user and course.
-        request.find_service(name="lti_h").sync([self.context.course], request.params)
+        self.request.find_service(name="lti_h").sync([self.context.course], lti_params)
 
         # Report all LTI assignment launches to the /reports page.
         LtiLaunches.add(
-            request.db,
-            request.params.get("context_id"),
-            request.params.get("oauth_consumer_key"),
+            self.request.db,
```

Comment:
I've seen this in another PR. It's fine to merge it with this if you want to avoid some chain-rebasing.

Reply:
Oh well. This should have already gone out and needed a rebase...